server/endpoint/logs/searcher/endpoint.go

```diff
@@ -0,0 +1,140 @@
+package searcher
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	kitendpoint "github.com/go-kit/kit/endpoint"
+	kithttp "github.com/go-kit/kit/transport/http"
+	"github.com/gorilla/mux"
+	"github.com/spf13/viper"
+
+	"github.com/giantswarm/prow-log-aggregator/flag"
+	"github.com/giantswarm/prow-log-aggregator/pkg/errors"
+	"github.com/giantswarm/prow-log-aggregator/server/middleware"
+	"github.com/giantswarm/prow-log-aggregator/service"
+)
+
+const (
+	// Method is the HTTP method this endpoint is registered for.
+	Method = "GET"
+	// Name identifies the endpoint. It is aligned to the package path.
+	Name = "logs"
+	// Path is the HTTP request path this endpoint is registered for.
+	Path = "/logs/{id}"
+)
+
+type Config struct {
+	Flag       *flag.Flag
+	Logger     micrologger.Logger
+	Middleware *middleware.Middleware
+	Service    *service.Service
+	Viper      *viper.Viper
+}
+
+type Endpoint struct {
+	logger     micrologger.Logger
+	middleware *middleware.Middleware
+	service    *service.Service
+
+	context    string
+	kubeconfig string
+	namespace  string
+}
+
+func New(config Config) (*Endpoint, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(errors.InvalidConfigError, "config.Logger must not be empty")
+	}
+	if config.Middleware == nil {
+		return nil, microerror.Maskf(errors.InvalidConfigError, "config.Middleware must not be empty")
+	}
+
+	e := &Endpoint{
+		logger:     config.Logger,
+		middleware: config.Middleware,
+		service:    config.Service,
+
+		context:    config.Viper.GetString(config.Flag.Kubeconfig.Context),
+		kubeconfig: config.Viper.GetString(config.Flag.Kubeconfig.Kubeconfig),
+		namespace:  config.Viper.GetString(config.Flag.Kubeconfig.Namespace),
+	}
+
+	return e, nil
+}
+
+func (e *Endpoint) Decoder() kithttp.DecodeRequestFunc {
+	return func(ctx context.Context, r *http.Request) (interface{}, error) {
+		vars := mux.Vars(r)
+		id := vars["id"]
+
+		return id, nil
+	}
+}
+
+func (e *Endpoint) Encoder() kithttp.EncodeResponseFunc {
+	return func(ctx context.Context, w http.ResponseWriter, response interface{}) error {
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
```

Comment:
Maybe this should be `text/plain` since the output of `tkn` isn't plain JSON.

Reply:
Done